### ISSUE TITLE
Develop Base Webserver

### ISSUE THREAD

**Issue by brycecorbitt** [ECE2305-Group-25/TPaaS#7]
Build scaffolding code for the TPaaS server. The server may be written with Python3's [http.server](https://docs.python.org/3/library/http.server.html) module, or any other 3rd party library such as [Flask](https://flask.palletsprojects.com/en/1.1.x/) or [Bottle](https://www.google.com/search?q=python3+bottle&oq=python3+bottle&aqs=chrome..69i57j0l7.1593j0j4&sourceid=chrome&ie=UTF-8). If http.server is used, boilerplate should be implemented in different module(s) to separate the request handling logic from that of processing the endpoints themselves.

This server should contain **all API endpoints** documented from #2, however logic for each endpoint does **not** need to be implemented. Endpoints should instead respond with a JSON which will be expanded upon when such logic is implemented. For now, the JSON only needs to contain the path to the endpoint and request type. 

Here is an example response:
```
{
    "endpoint": "/api/my_custom_endpoint",
    "request_type": "GET"
}
```

**Comment by nhhollander** [comment]
As far as selection of a backed goes, there are a few important parameters we need to keep in mind.  Specifically, I'm concerned about how we're going to handle relaying a message from the server to the dispenser.

Originally I wanted to use Python's built in `http.server` module because it spawns threads within a larger application allowing a single constant connection to the dispenser to be shared by all requests.  Unfortunately `http.server` has several fatal security flaws that render it unsuitable to our project.

Right now I'm looking into flask and seeing how it handles requests.  At a minimum I know flask has some sort of mechanism for IPC built in that can be used for this kind of situation.

I'll also look into defining a dispenser ⇆ server interface/protocol.

**Comment by brycecorbitt** [comment]
In the implementation of our Design Proposal, I defined the web server with the intent that it will be running on the Raspberry Pi itself. Developing a server to manage one or more dispensers would be a necessity at a commercial level. However, we don't need to implement one for our prototype. To keep server code as simple as simple as possible, a single webserver that utilize the dispenser controls directly at each API endpoint will be sufficient for our project. 

While it's a rather crude example, I'm envisioning our webserver being implemented similarly to a test server I wrote [here](https://github.com/brycecorbitt/MU2801_Percussion/blob/master/web_server.py) (minus the HTML) for controlling a motor to hit a snare drum. Modules are written to control each component, and are then are abstracted into modules that define operations to be carried out by the components (similar to how functionality of the "ADC" and "DCDriver" classes are used in the "PercussionMotor" class). After these modules are written, they are imported to the webserver where they are used to control the components based on the purpose of each endpoint.

All things considered, I think Flask is a good way to go for implementing this webserver. It will certainly help us streamline development. There's some documentation to handling HTTP requests [here](https://flask.palletsprojects.com/en/1.1.x/quickstart/#http-methods) that I think will be helpful. Flask also has a builtin function for [preparing JSON responses](https://flask.palletsprojects.com/en/1.1.x/api/#flask.json.jsonify), which could be of use as well.
